### ISSUE TITLE
Spack compiler wrappers aren't compatible with SCons

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#4927]
I'm working on a `SConsPackage` base class and started debugging some SCons builds. I discovered that SCons ignores the `CC` environment variable. If I add `'CC={0}'.format(spack_cc)` to the `serf` build line, it actually uses Spack's compiler wrappers, but I end up seeing the following error message:
```
Spack compiler must be run from Spack! Input 'SPACK_PREFIX' is missing.
```
Maybe SCons purifies the build environment for some reason? We might need a way to work around this. For now I'll tell it to use the actual compilers and manually set `-L` and `-I`.

**Comment by adamjstewart** [comment]
Dug a little deeper into this. It looks like SCons runs all build recipes in an isolated Python environment. By default, this environment doesn't pass any environment variables from the parent process to the subprocess (even `PATH`!). It is up to the package developers to manually pass these environment variables. Some packages like `cantera` have a `env_vars=all` option to pass these environment variables, but others like `serf` don't. I don't think there will be a generic way to fix this.

**Comment by hyoklee** [comment]
@alalazo , any progress on this issue?

**Comment by alalazo** [comment]
Not that I know of. Are you running into the same problem? What is the spec you are trying to build?

**Comment by hyoklee** [comment]
Yes. I'm building [DAOS spec](https://github.com/hyoklee/spack/tree/develop/var/spack/repos/builtin/packages/daos).

**Comment by adamjstewart** [comment]
We managed to patch bazel to pass in Spack's env vars. I wonder if we could do the same for SCons.
